test/OpenTelemetry.Instrumentation.Runtime.Tests/RuntimeMetricsTests.cs

```diff
@@ -34,9 +37,156 @@ public void RuntimeMetricsAreCaptured()
+            Assert.NotNull(gcCountMetric);
+
+#if NETCOREAPP3_1_OR_GREATER
+            var gcAllocationSizeMetric = exportedItems.FirstOrDefault(i => i.Name == "process.runtime.dotnet.gc.allocations.size");
+            Assert.NotNull(gcAllocationSizeMetric);
+#endif
+
+#if NET6_0_OR_GREATER
+            var gcCommittedMemorySizeMetric = exportedItems.FirstOrDefault(i => i.Name == "process.runtime.dotnet.gc.committed_memory.size");
+            Assert.NotNull(gcCommittedMemorySizeMetric);
+
+            var gcHeapSizeMetric = exportedItems.FirstOrDefault(i => i.Name == "process.runtime.dotnet.gc.heap.size");
+            Assert.NotNull(gcHeapSizeMetric);
+
+            if (Environment.Version.Major >= 7)
+            {
+                var gcHeapFragmentationSizeMetric = exportedItems.FirstOrDefault(i => i.Name == "process.runtime.dotnet.gc.heap.fragmentation.size");
+                Assert.NotNull(gcHeapFragmentationSizeMetric);
+            }
+#endif
+        }
+
+#if NET6_0_OR_GREATER
+        [Fact]
+        public void JitRelatedMetricsTest()
+        {
+            var exportedItems = new List<Metric>();
+            using var meterProvider = Sdk.CreateMeterProviderBuilder()
+                 .AddRuntimeInstrumentation()
+                 .AddInMemoryExporter(exportedItems)
+                .Build();
+
+            meterProvider.ForceFlush(MaxTimeToAllowForFlush);
+
+            var jitCompiledSizeMetric = exportedItems.FirstOrDefault(i => i.Name == "process.runtime.dotnet.jit.il_compiled.size");
+            Assert.NotNull(jitCompiledSizeMetric);
+
+            var jitMethodsCompiledCountMetric = exportedItems.FirstOrDefault(i => i.Name == "process.runtime.dotnet.jit.methods_compiled.count");
+            Assert.NotNull(jitMethodsCompiledCountMetric);
+
+            var jitCompilationTimeMetric = exportedItems.FirstOrDefault(i => i.Name == "process.runtime.dotnet.jit.compilation_time");
+            Assert.NotNull(jitCompilationTimeMetric);
+        }
+#endif
+
+#if NETCOREAPP3_1_OR_GREATER
+        [Fact]
+        public void ThreadingRelatedMetricsTest()
+        {
+            var exportedItems = new List<Metric>();
+            using var meterProvider = Sdk.CreateMeterProviderBuilder()
+                 .AddRuntimeInstrumentation()
+                 .AddInMemoryExporter(exportedItems)
+                .Build();
+
+            meterProvider.ForceFlush(MaxTimeToAllowForFlush);
+
+            var lockContentionCountMetric = exportedItems.FirstOrDefault(i => i.Name == "process.runtime.dotnet.monitor.lock_contention.count");
+            Assert.NotNull(lockContentionCountMetric);
+
+            var threadCountMetric = exportedItems.FirstOrDefault(i => i.Name == "process.runtime.dotnet.thread_pool.threads.count");
+            Assert.NotNull(threadCountMetric);
+
+            var completedItemsCountMetric = exportedItems.FirstOrDefault(i => i.Name == "process.runtime.dotnet.thread_pool.completed_items.count");
```

Comment:
its possible to cause this metric to be atleast N, if you queue N items to TP, and wait for it to complete. 
Similar to  Exception.Count
Timer.Count etc.

Not blocker for the PR.

Reply:
Addressed in #559.